classes/lifecycle/course_table.php

```diff
@@ -33,43 +39,50 @@ class course_table extends \table_sql
      *
      * @throws \coding_exception
      */
-    public function __construct($filterdata = [])
-    {
+    public function __construct($filterdata = []) {
         global $DB;
 
         parent::__construct('tool_lcbackupcoursestep-course');
 
-        // Action buttons string
+        // Action buttons string.
         $this->strings = [
             'download' => get_string('download'),
             'restore' => get_string('restore'),
         ];
 
         // Build the SQL.
         $fields = 'f.id as id,
-                   co.id as courseid, co.shortname as courseshortname, co.fullname as coursefullname,
-                   f.filename as filename, f.filesize as filesize, f.timecreated as createdat';
+            COALESCE(co.id, md.oldcourseid) as courseid,
+            COALESCE(co.shortname, md.shortname) as courseshortname,
+            COALESCE(co.fullname, md.fullname) as coursefullname,
+            f.filename as filename, f.filesize as filesize, f.timecreated as createdat';
         $from = '{files} f
-                 JOIN {context} c ON c.id = f.contextid
-                 JOIN {course} co ON co.id = c.instanceid';
+            JOIN {context} c ON c.id = f.contextid
+            LEFT JOIN {course} co ON co.id = c.instanceid
+            LEFT JOIN {tool_lcbackupcoursestep_metadata} md ON f.id = md.fileid';
 
         $where = ["f.component = :component AND filename <> '.'"];
         $params = ['component' => 'tool_lcbackupcoursestep'];
 
+        // Check for both course context and system context.
+        $where[] = "(c.contextlevel = :coursecontext OR c.contextlevel = :systemcontext)";
+        $params['coursecontext'] = CONTEXT_COURSE;
+        $params['systemcontext'] = CONTEXT_SYSTEM;
+
```

Comment:
The backup file is is no longer stored in the context of course. Why are we checking the context for course? If we need to check against both, please add that to the comment why we are checking for both.

Reply:
The backup file is no longer stored in the course context, but older course backups were stored there. If we remove the check for the course context, the old backups won’t be visible. To be safe and keep both old and new backups accessible, I added the check for both course and system contexts. Will add this to the description.

Reply:
I think this should happen in the upgrade script. The upgrade script would move any backups that are currently in the course context to system context, and also populate the new table. Note that when populating the new table, the timestamp should be the created at time of the backup file. This would also mean we can remove coalesce in the queries. This will reduce the maintenance.  

 - [ ] Add to the upgrade.php moving the current backup files from course context to system context
 - [ ] Populate the new table with the moved files.
 - [ ] Remove any references of course context from the changes, no more coalesce should be needed.